cql3/restrictions/statement_restrictions.cc

```diff
@@ -174,7 +202,11 @@ static std::vector<expr::expression> extract_partition_range(
                         found->second = make_conjunction(std::move(found->second), b);
                     }
                 }
-            }
+            });
+        }
+
+        void operator()(const column_value_tuple& s) {
+            // Ignore tuples while evaluating single-column restrictions
```

Comment:
We're not limited to single-column restrictions here, but to partition-key restrictions (including token).  The real reason for ignoring tuples is that the parser disallows partition columns in tuples.

Reply:
Is it the parser or something lower? Either way I'll adjust.